src/core-services/settings/mutations/updateAppSettings.js

```diff
@@ -24,10 +24,10 @@ export default async function updateAppSettings(context, settingsUpdates, shopId
   }
 
   // Look up roles that are allowed to set each setting. Throw if not allowed.
-  updateKeys.forEach((field) => {
+  updateKeys.forEach(async (field) => {
     const allowedRoles = shopId ? rolesThatCanEditShopSetting(field) : rolesThatCanEditGlobalSetting(field);
-    if (allowedRoles.length === 0 || !userHasPermission(allowedRoles, shopId)) {
-      throw new ReactionError("access-denied", `You are not allowed to edit the "${field}" setting`);
+    if (allowedRoles.length === 0) {
+      await checkPermissions(allowedRoles, shopId);
```

Comment:
Use `for...of`

Reply:
Also, you need to keep the `allowedRoles.length === 0` check but also do the `checkPermissions` check always.

```js
if (allowedRoles.length === 0) {
  throw new ReactionError("access-denied", `You are not allowed to edit the "${field}" setting`);
}
await checkPermissions(allowedRoles, shopId);
```